pyomo/contrib/doe/fim_doe.py

```diff
@@ -623,6 +641,9 @@ def compute_FIM(self, design_values, mode='sequential_finite', FIM_store_name=No
             # call generator function to get scenario dictionary
             scena_gen = Scenario_generator(self.param_init, formula=self.formula, step=self.step)
             scena_gen.generate_sequential_para()
+            scena_set = scena_gen.simultaneous_scenario()
+            self.scena_set = scena_set
+            print(self.scena_set)
 
             # if measurements are provided
             if read_output is not None:
```

Comment:
This code seems to be duplicated in the function. How can we refactor to remove the duplication?